packages/be/crons/cid-importer.js

```diff
@@ -0,0 +1,231 @@
+ *
+ * Caches CID data
+ *
+ */
+
+// ///////////////////////////////////////////////////// Imports + general setup
+// -----------------------------------------------------------------------------
+const ModuleAlias = require('module-alias')
+const Path = require('path')
+const Axios = require('axios')
+const Fs = require('fs-extra')
+const Express = require('express')
+const Util = require('util')
+const Stream = require('stream')
+const Pipeline = Util.promisify(Stream.pipeline)
+const Spawn = require('child_process').spawn
+require('dotenv').config({ path: Path.resolve(__dirname, '../.env') })
+
+const MC = require('../config')
+
+// ////////////////////////////////////////////////////////////////// Initialize
+MC.app = Express()
+
+// ///////////////////////////////////////////////////////////////////// Aliases
+ModuleAlias.addAliases({
+  '@Root': MC.packageRoot,
+  '@Modules': `${MC.packageRoot}/modules`
+})
+
+try {
+  const modulesRoot = `${MC.packageRoot}/modules`
+  const items = Fs.readdirSync(modulesRoot)
+  items.forEach((name) => {
+    const path = `${modulesRoot}/${name}`
+    if (Fs.statSync(path).isDirectory()) {
+      const moduleName = `${name[0].toUpperCase() + name.substring(1)}`
+      ModuleAlias.addAlias(`@Module_${moduleName}`, path)
+    }
+  })
+} catch (e) {
+  console.log(e)
+}
+
+// ///////////////////////////////////////////////////////////////////// Modules
+require('@Module_Database')
+require('@Module_Cid')
+const { GetFileFromDisk } = require('@Module_Utilities')
+
+// /////////////////////////////////////////////////////////////////// Functions
+// ------------------------------------------------------------ retrieveCidFiles
+const retrieveCidFiles = async (uploads) => {
+  try {
+    const len = uploads.length
+    const cidFileRefs = []
+    for (let i = 0; i < len; i++) {
+      const upload = uploads[i]
+      // if a file already exists with this name in the temp folder, 
+      // delete it to make way for an updated version
+      await deleteTemporaryFile(upload.name) 
+      // fetch file using upload cid
+      const response = await Axios.get(`https://${upload.cid}.ipfs.w3s.link/`, {
+        responseType: 'stream'
+      })
+      await Pipeline(response.data, Fs.createWriteStream(`${MC.packageRoot}/tmp/cid-files/${upload.name}`))
+      cidFileRefs.push({
+        cid: upload.cid,
+        name: upload.name
+      })
+    }
+    return await unpackRetrievedFiles(cidFileRefs)
+  } catch (e) {
+    console.log('====================================== [Function: unpackCids]')
+    console.log(e)
+    process.exit(0)
+  }
+}
+
+// --------------------------------------------------------------- unpackZstFile
+const unpackZstFile = (file) => {
+  return new Promise((resolve, reject) => {
+    const unzstd = Spawn('unzstd', [`../tmp/cid-files/${file.name}`])
+    const errors = []
+    unzstd.stderr.on('data', (msg) => {
+      errors.push(`Error unpacking ${file.name}: ${msg.toString()}`)
```

Comment:
This will catch errors, but should we also have a checker for file name? i.e., if the file name doesn't end with `.json.zst`, skip it. 

This would exclude our sample uploads, and also accidental uploads from the user.

Reply:
@orvn there already is this check on line 106

Reply:
@svvimming currently your checker is inside `unpackRetrievedFiles` but it should be further up the chain. Meaning, if it doesn't end with `.json.zst` then don't download it at all!